### PR TITLE
Experimental Firefox Support

### LINKED ISSUE #1472: Set default Firefox prefs
https://github.com/kblok/puppeteer/pull/20/commits/35d5ba5830402af16070cafdc071c1970875a0d9

### LINKED ISSUE #1494: Test firefox in CI

### LINKED ISSUE #1617: puppeteer-sharp downloads old version of chromium
### Description
When puppeteer runs (windows 10) it downloads Win64-706915 version of chrome

### Complete minimal example reproducing the issue
Get latest nuget of puppeteer-sharp and run any code that will cause the download of chromium
Observe the above version is being downloaded (from 10/17/2019)

### PR DESCRIPTION
**Pull request by kblok**
closes #1472 
closes #1494
closes #1617

### CODE REVIEW

**Comment by slashP** [comment]
I'm not sure how far this work has come or if you want input, but when I try it out locally on Windows it fails with 

```
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at PuppeteerSharp.States.ChromiumStartingState.<StartCoreAsync>d__6.MoveNext() in C:\dev\priv\puppeteer-sharp\lib\PuppeteerSharp\States\ChromiumStartingState.cs:line 82
```

after initializing with `Product.Firefox` in `BrowserFetcherOptions` and `LaunchOptions`. The error occurs first, then the browser does in fact start. Looking forward to seeing this working!

**Comment by kblok** [comment]
Hey @slashP what's the exception?

**Comment by slashP** [comment]
`PuppeteerSharp.ProcessException: 'Failed to launch browser!` on `await p.StartCompletionSource.Task.ConfigureAwait(false);` from the `OnProcessExitedWhileStarting` method it looks like. Process has start info:

```
StartInfo	FileName="C:\\dev\\gts\\Gts.Esea.UiTests\\src\\Gts.Esea.UiTests\\bin\\Debug\\net5.0\\.local-firefox\\Win64-87.0a1\\firefox\\firefox.exe", Arguments="--no-remote --foreground about:blank --remote-debugging-port=0 --profile \"C:\\Users\\perkr\\AppData\\Local\\Temp\\czwttcyf.iu3\"", WorkingDirectory=""
```

Not sure how to extract more info from it at this point.

**Comment by kblok** [comment]
It seems that the project is not building.

File: appveyor.yml [hunk 1]
```diff
@@ -14,10 +14,6 @@ environment:
       ENQUEUE_ASYNC_MESSAGES: true
     - framework: netcoreapp2.2
       ENQUEUE_ASYNC_MESSAGES: true
-    - framework: net48
-      ENQUEUE_ASYNC_MESSAGES: false
-    - framework: netcoreapp2.2
-      ENQUEUE_ASYNC_MESSAGES: false
```
[the remark this answers is not in the document]
Comment (Androbin) [reply appveyor.yml:20]: Speaking of which, is there any good reason to set `EnqueueTransportMessages = False` ?

[the remark this answers is not in the document]
Comment (kblok) [reply appveyor.yml:20]: It doesn't make much sense to test all the combinations with `ENQUEUE_ASYNC_MESSAGES`.  
Puppeteer works well in .NET Core without enqueuing messages. That was implemented for .NET Framework users.

**Pull request merged**